src/duckling/game/collision/collision-types.service.ts

```diff
@@ -93,13 +97,13 @@ export class CollisionTypesService {
         this._snackbar.addSnack(message);
     }
 
-    private _findUnknownCollisionTypes(map : RawMapFile) : string[] {
+    private _findUnknownCollisionTypes(map : RawMapFile) : Set<string> {
         let rawMapCollisionTypes = this._getUniqueCollisionTypesInRawMapFile(map);
-        let unknownCollisionTypes : string[] = [];
+        let unknownCollisionTypes = new Set<string>([]);
         if (rawMapCollisionTypes) {
             for (let type of rawMapCollisionTypes) {
-                if (this.collisionTypes.getValue().indexOf(type) === -1) {
-                    unknownCollisionTypes.push(type);
+                if (!this.collisionTypes.getValue().has(type)) {
```

Comment:
Yes it is because the "unknown" collision types are located by finding ones present in the map file that aren't present in the ones that are loaded in already to this.collisionTypes